+= on private property doesn't count as a read for --noUnusedLocals

<!-- 🚨 STOP 🚨 𝗦𝗧𝗢𝗣 🚨 𝑺𝑻𝑶𝑷 🚨

Half of all issues filed here are duplicates, answered in the FAQ, or not appropriate for the bug tracker. Even if you think you've found a *bug*, please read the FAQ first, especially the Common "Bugs" That Aren't Bugs section!

Please help us by doing the following steps before logging an issue:
  * Search: https://github.com/Microsoft/TypeScript/search?type=Issues
  * Read the FAQ: https://github.com/Microsoft/TypeScript/wiki/FAQ

Please fill in the *entire* template below.
-->

<!-- Please try to reproduce the issue with `typescript@next`. It may have already been fixed. -->
**TypeScript Version:**  2.9.2

<!-- Search terms you tried before logging this (so others can find this issue more easily) -->
**Search Terms:** 
--noUnusedLocals
+=
private property

**Code**
**tsconfig.json**:
```
{
   "compileOnSave": true,
   "compilerOptions": {
      "target": "es6",
      "module": "none",
      "noImplicitAny": true,
      "removeComments": false,
      "outFile": "test.js",
      "declaration": false,
      "noEmitOnError": true,
      "sourceMap": false,
      "strictNullChecks": true,
      "noUnusedLocals": true,
      "noUnusedParameters": true
   }
}
```
**test.ts**
```
export class TestClass {
   private mNumber: number = 0;
   private get Number(): number { return this.mNumber; }
   private set Number(value: number) { this.mNumber = value; }

   public DoSomething() {
      this.Number += 1;
   }
}
```

**Result**
```
c:\Temp>tsc
test.ts(3,16): error TS6133: 'Number' is declared but its value is never read.
```

**Expected behavior:**
+= should count as both a read and a write

**Actual behavior:**
+= seems to count as a write but not a read. though this problem doesn't occur with a public property.

**Playground Link:** <!-- A link to a TypeScript Playground "Share" link which demonstrates this behavior --> 
I don't think I can set --noUnusedLocals on the playground

**Related Issues:** <!-- Did you find other bugs that looked similar? -->
I found this "noUnusedLocals checks don't work with private modifier": https://github.com/Microsoft/TypeScript/issues/15775
but it was closed

--- a/src/compiler/checker.ts
+++ b/src/compiler/checker.ts
@@ -23535,7 +23535,7 @@ namespace ts {
             if (!hasPrivateModifier && !hasPrivateIdentifier) {
                 return;
             }
-            if (nodeForCheckWriteOnly && isWriteOnlyAccess(nodeForCheckWriteOnly) && !(prop.flags & SymbolFlags.SetAccessor && !(prop.flags & SymbolFlags.GetAccessor))) {
+            if (nodeForCheckWriteOnly && isWriteOnlyAccess(nodeForCheckWriteOnly) && !(prop.flags & SymbolFlags.SetAccessor)) {
                 return;
             }
 

--- a/tests/baselines/reference/unusedPrivateMembers.errors.txt
+++ b/tests/baselines/reference/unusedPrivateMembers.errors.txt
@@ -1,0 +1,70 @@
+tests/cases/compiler/unusedPrivateMembers.ts(56,13): error TS6133: 'b' is declared but its value is never read.
+
+
+==== tests/cases/compiler/unusedPrivateMembers.ts (1 errors) ====
+    class Test1 {
+        private initializeInternal() {
+        }
+    
+        public test() {
+            var x = new Test1();
+            x.initializeInternal();
+        }
+    }
+    
+    class Test2 {
+        private p = 0;
+        public test() {
+            var x = new Test2();
+            x.p;
+        }
+    }
+    
+    class Test3 {
+        private get x () {
+            return 0;
+        }
+    
+        public test() {
+            var x = new Test3();
+            x.x;
+        }
+    }
+    
+    class Test4 {
+        private set x(v) {
+            v;
+        }
+    
+        public test() {
+            var x = new Test4();
+            x.x;
+        }
+    }
+    
+    class Test5<T> {
+        private p: T;
+        public test() {
+            var x = new Test5<number>();
+            x.p;
+        }
+    }
+    
+    class Test6 {
+        private get a() {
+            return 0;
+        }
+        private set a(v) {
+            v;
+        }
+        private b = 0;
+                ~
+!!! error TS6133: 'b' is declared but its value is never read.
+    
+        public test() {
+            var x = new Test6();
+            x.a++;
+            x.b++;
+        }
+    }
+    

--- a/tests/baselines/reference/unusedPrivateMembers.js
+++ b/tests/baselines/reference/unusedPrivateMembers.js
@@ -47,6 +47,22 @@ class Test5<T> {
     }
 }
 
+class Test6 {
+    private get a() {
+        return 0;
+    }
+    private set a(v) {
+        v;
+    }
+    private b = 0;
+
+    public test() {
+        var x = new Test6();
+        x.a++;
+        x.b++;
+    }
+}
+
 
 //// [unusedPrivateMembers.js]
 var Test1 = /** @class */ (function () {
@@ -110,4 +126,25 @@ var Test5 = /** @class */ (function () {
         x.p;
     };
     return Test5;
+}());
+var Test6 = /** @class */ (function () {
+    function Test6() {
+        this.b = 0;
+    }
+    Object.defineProperty(Test6.prototype, "a", {
+        get: function () {
+            return 0;
+        },
+        set: function (v) {
+            v;
+        },
+        enumerable: true,
+        configurable: true
+    });
+    Test6.prototype.test = function () {
+        var x = new Test6();
+        x.a++;
+        x.b++;
+    };
+    return Test6;
 }());

--- a/tests/baselines/reference/unusedPrivateMembers.symbols
+++ b/tests/baselines/reference/unusedPrivateMembers.symbols
@@ -110,3 +110,40 @@ class Test5<T> {
     }
 }
 
+class Test6 {
+>Test6 : Symbol(Test6, Decl(unusedPrivateMembers.ts, 46, 1))
+
+    private get a() {
+>a : Symbol(Test6.a, Decl(unusedPrivateMembers.ts, 48, 13), Decl(unusedPrivateMembers.ts, 51, 5))
+
+        return 0;
+    }
+    private set a(v) {
+>a : Symbol(Test6.a, Decl(unusedPrivateMembers.ts, 48, 13), Decl(unusedPrivateMembers.ts, 51, 5))
+>v : Symbol(v, Decl(unusedPrivateMembers.ts, 52, 18))
+
+        v;
+>v : Symbol(v, Decl(unusedPrivateMembers.ts, 52, 18))
+    }
+    private b = 0;
+>b : Symbol(Test6.b, Decl(unusedPrivateMembers.ts, 54, 5))
+
+    public test() {
+>test : Symbol(Test6.test, Decl(unusedPrivateMembers.ts, 55, 18))
+
+        var x = new Test6();
+>x : Symbol(x, Decl(unusedPrivateMembers.ts, 58, 11))
+>Test6 : Symbol(Test6, Decl(unusedPrivateMembers.ts, 46, 1))
+
+        x.a++;
+>x.a : Symbol(Test6.a, Decl(unusedPrivateMembers.ts, 48, 13), Decl(unusedPrivateMembers.ts, 51, 5))
+>x : Symbol(x, Decl(unusedPrivateMembers.ts, 58, 11))
+>a : Symbol(Test6.a, Decl(unusedPrivateMembers.ts, 48, 13), Decl(unusedPrivateMembers.ts, 51, 5))
+
+        x.b++;
+>x.b : Symbol(Test6.b, Decl(unusedPrivateMembers.ts, 54, 5))
+>x : Symbol(x, Decl(unusedPrivateMembers.ts, 58, 11))
+>b : Symbol(Test6.b, Decl(unusedPrivateMembers.ts, 54, 5))
+    }
+}
+

--- a/tests/baselines/reference/unusedPrivateMembers.types
+++ b/tests/baselines/reference/unusedPrivateMembers.types
@@ -116,3 +116,45 @@ class Test5<T> {
     }
 }
 
+class Test6 {
+>Test6 : Test6
+
+    private get a() {
+>a : number
+
+        return 0;
+>0 : 0
+    }
+    private set a(v) {
+>a : number
+>v : number
+
+        v;
+>v : number
+    }
+    private b = 0;
+>b : number
+>0 : 0
+
+    public test() {
+>test : () => void
+
+        var x = new Test6();
+>x : Test6
+>new Test6() : Test6
+>Test6 : typeof Test6
+
+        x.a++;
+>x.a++ : number
+>x.a : number
+>x : Test6
+>a : number
+
+        x.b++;
+>x.b++ : number
+>x.b : number
+>x : Test6
+>b : number
+    }
+}
+

--- a/tests/cases/compiler/unusedPrivateMembers.ts
+++ b/tests/cases/compiler/unusedPrivateMembers.ts
@@ -49,3 +49,19 @@ class Test5<T> {
         x.p;
     }
 }
+
+class Test6 {
+    private get a() {
+        return 0;
+    }
+    private set a(v) {
+        v;
+    }
+    private b = 0;
+
+    public test() {
+        var x = new Test6();
+        x.a++;
+        x.b++;
+    }
+}

